src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyStringComparison.cs

```diff
@@ -81,6 +86,19 @@ public override void Initialize(AnalysisContext analysisContext)
                         return;
                     }
 
+                    if (linqExpressionType != null)
+                    {
+                        var enclosingLambdaOrLocalFunc = invocationExpression.GetAncestor(s_LambdaOrLocalFunctionKinds);
+
+                        // Check if we are in a Expression<Func<T...>> context, in which case it is possible
```

Comment:
Rather than `IQueryable`, I am now checking for `Expression<TDelegate>` (as suggested by @damieng).